### PR TITLE
Fix type hints for AnimationLibrary's signals' arguments

### LINKED ISSUE #61960: AnimationLibrary's signals' type hints are wrong
### Godot version

v4.0.alpha.custom_build [1f690f197]

### System information

Linux (Fedora 36)

### Issue description

All of the signals on AnimationLibrary have arguments, all of which are of type StringName, however the type hint for them is Animation.
```gdscript
# connect the animation_added signal of an AnimationLibrary to this:
func _on_animation_library_animation_added(name):
   print(name is StringName)
   print(name is Animation)
```
This prints `true`, `false`, as it should, but according to these type hints it should be `false`, `true`.

### Steps to reproduce

1. Open the documentation.
2. Look up AnimationLibrary and look at the signals.
3. They all show Animation as the type of their arguments when it should be StringName.

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by jtnicholl**
Fixes #61960.

### CODE REVIEW

File: scene/resources/animation_library.cpp [hunk 1]
```diff
@@ -143,9 +143,9 @@ void AnimationLibrary::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("_get_data"), &AnimationLibrary::_get_data);
 
 	ADD_PROPERTY(PropertyInfo(Variant::DICTIONARY, "_data", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_NO_EDITOR), "_set_data", "_get_data");
-	ADD_SIGNAL(MethodInfo("animation_added", PropertyInfo(Variant::OBJECT, "name", PROPERTY_HINT_RESOURCE_TYPE, "Animation")));
-	ADD_SIGNAL(MethodInfo("animation_removed", PropertyInfo(Variant::OBJECT, "name", PROPERTY_HINT_RESOURCE_TYPE, "Animation")));
-	ADD_SIGNAL(MethodInfo("animation_renamed", PropertyInfo(Variant::OBJECT, "name", PROPERTY_HINT_RESOURCE_TYPE, "Animation"), PropertyInfo(Variant::OBJECT, "to_name", PROPERTY_HINT_RESOURCE_TYPE, "Animation")));
+	ADD_SIGNAL(MethodInfo("animation_added", PropertyInfo(Variant::OBJECT, "name", PROPERTY_HINT_RESOURCE_TYPE, "StringName")));
+	ADD_SIGNAL(MethodInfo("animation_removed", PropertyInfo(Variant::OBJECT, "name", PROPERTY_HINT_RESOURCE_TYPE, "StringName")));
+	ADD_SIGNAL(MethodInfo("animation_renamed", PropertyInfo(Variant::OBJECT, "name", PROPERTY_HINT_RESOURCE_TYPE, "StringName"), PropertyInfo(Variant::OBJECT, "to_name", PROPERTY_HINT_RESOURCE_TYPE, "StringName")));
```
Comment (timothyqiu) [inline scene/resources/animation_library.cpp:148]: `StringName` is a distinct type `Variant::STRING_NAME`. For example:

https://github.com/godotengine/godot/blob/d4235f51b1a88e48d676b877422fb121d326935c/scene/animation/animation_player.cpp#L2103

Comment (jtnicholl) [reply scene/resources/animation_library.cpp]: My bad, fixed it.

File: scene/resources/animation_library.cpp [hunk 2]
```diff
@@ -143,9 +143,9 @@ void AnimationLibrary::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("_get_data"), &AnimationLibrary::_get_data);
 
 	ADD_PROPERTY(PropertyInfo(Variant::DICTIONARY, "_data", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_NO_EDITOR), "_set_data", "_get_data");
-	ADD_SIGNAL(MethodInfo("animation_added", PropertyInfo(Variant::OBJECT, "name", PROPERTY_HINT_RESOURCE_TYPE, "Animation")));
-	ADD_SIGNAL(MethodInfo("animation_removed", PropertyInfo(Variant::OBJECT, "name", PROPERTY_HINT_RESOURCE_TYPE, "Animation")));
-	ADD_SIGNAL(MethodInfo("animation_renamed", PropertyInfo(Variant::OBJECT, "name", PROPERTY_HINT_RESOURCE_TYPE, "Animation"), PropertyInfo(Variant::OBJECT, "to_name", PROPERTY_HINT_RESOURCE_TYPE, "Animation")));
+	ADD_SIGNAL(MethodInfo("animation_added", PropertyInfo(Variant::STRING_NAME, "name")));
+	ADD_SIGNAL(MethodInfo("animation_removed", PropertyInfo(Variant::OBJECT, "name")));
```
Comment (akien-mga) [inline scene/resources/animation_library.cpp:147]: You missed this one.

Comment (jtnicholl) [reply scene/resources/animation_library.cpp]: Ok, I got them all now.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**